### PR TITLE
Added more release notes/changelog options when uploading to Beta

### PR DESCRIPTION
**Pull request by dtrenz**
Previously, the only option for release notes uploaded for Beta builds
was to use the git commit log (since last tag).

I've added 2 more options for Beta release notes:
  1. Added the option to use the Jenkins changes (since last builg) as changelog for release notes.
  2. Added the ability to pass a string to be used as the release notes (e.g. "PR #123").

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -54,16 +54,31 @@ platform :ios do
   end
   
   desc "Upload a local IPA to Crashlytics with (group)."
+  desc "#### Options"
+  desc " * **`generate_changelog`**: Whether or not to generate a changelog as the release notes (boolean – default: `false`)"
+  desc " * **`changelog_source`**: The source to use to generate the changelog (e.g. `\"git\"`, `\"jenkins\"` – default: `"git"`)"
```
Comment (jshier) [inline fastlane/Fastfile]: I think we need some additional options here. For some projects, the git changes may pick up so many commits that the maximum changelong length is exceeded. For those, we've just been using the PR name, which is a safe default. You can take a look at the DTE Fastfile to see how this is generated.

**Comment by dtrenz** [comment]
@jshier I added a PR build changelog option. Let me know what you think.

File: fastlane/Fastfile [hunk 2]
```diff
@@ -54,16 +54,39 @@ platform :ios do
...
-      changelog_from_git_commits
+
+    unless options.key?(:release_notes)
+      if options[:generate_changelog] == true
+        case options[:changelog_type]
+          when "jenkins"
+            make_changelog_from_jenkins
+
+          when "pr"
+            UI.error("uploadToCrashlytics: Environment variable for PR ID is not set (ghprbPullId).") unless ENV.key?("ghprbPullId")
```
Comment (jshier) [inline fastlane/Fastfile]: Wasn't the `ENV.key?` syntax added in a fairly recent version of Ruby? I'm fine with adding it, but there may need to be something which checks the version of Ruby being called. I could be wrong though.

Comment (dtrenz) [reply fastlane/Fastfile]: It looks like this has been available since ruby 2.1.4: https://ruby-doc.org/core-2.1.4/ENV.html#method-c-key-3F

**Pull request merged**